connection.go

```diff
@@ -740,6 +740,14 @@ func (c *Connection) handleFrameNoRelay(frame *Frame) bool {
 // writeFrames is the main loop that pulls frames from the send channel and
 // writes them to the connection.
 func (c *Connection) writeFrames(_ uint32) {
+	defer func() {
```

Comment:
would it make sense to move this defer inside `connectionError` function?

It feels like `writeFrames` is about sending data, while `connectionError` is about "closing" and cleanup?

It looks like `connectionError` is used in a bunch of other places too - so perhaps it would be more generally useful?

Reply:
Or, perhaps, even in `closeNetwork`? 

This would nicely tie in "destruction" of the object with releasing the frames?